### DISCUSSION TITLE
Any way to use folders as books without to import first?

### DISCUSSION THREAD

**Discussion by hdo** [TortugaPower/BookPlayer#1069] [Ideas]
I'm about to migrate back to iOS after a couple of year.
On Android I'm use Simple Audiobook Player (https://play.google.com/store/apps/details?id=mdmt.sabp.free) for
listening to my audiobooks. 
This application is extremly simple to use and is very fast with even large set of audiobooks.
I don't need to import files first before listening to them.

With BookPlayer I noticed I could copy files to the 'shared' area of the application however
this does not work without importing them first.

The suggested way is to put all files of an audiobook into a ZIP archive and import them.

Is there are plans to support audiobook folders without importing them first?

E.g. with the iPhone 15 I could transfer all my current audiobook folders to the 'shared' area
of BookPlayer via USB-C cable. I really don't want to import the files within the app again.
Maybe it could be feasible to add something like 'rescan'. 
There is a 'repair' function but this does not seem to respect the folder structure :-(

Any hints or ideas?

**Comment by GianniCarlo** [comment]
Hi @hdo , I'm not sure I follow entirely, there shouldn't be a need of the 'repair' function at all when importing, when you say you move the folders to the 'shared' area, do you mean inside the 'Processed' folder? if transferring directly, you should put the files/folders inside the app's Documents folder, at the same level as the Processed folder, and when you next open the app, it will prompt to process those items, let me know if I'm missing something or misunderstanding the scenario.

Hmmmm I think I get your point, do you mean when transferring inside the Processed folder, for the app to automatically create the items, and well in that case we wouldn't really need a Processed folder, and just use the Documents folder of the app 🤔, I have to think a bit more on this, as the Processed folder was something from v1, where we hashed the file to get an unique identifier for it as it was all stored in a flat structure, and needed a separate space to know which ones were pending, and as the app continued to grow, we ditched that in favor of doing the same file hierarchy on disk as on the DB, and made use of the separation so it's clear for the user what they are importing, and for us to create the DB representation of the file and extracting the metadata (which if we do the change, will still happen for files transferred via Airdrop or Files app integration as those are placed by the system in an 'Inbox' folder and have to be moved, and well also from the shared extension which are put in a shared container and also need to be moved)

**Reply by hdo** [reply]
> you should put the files/folders inside the app's Documents folder, at the same level as the Processed folder, and when you next open the app, it will prompt to process those items, let me know if I'm missing something or misunderstanding the scenario.

Yeah my bad. I somehow managed to 'corrupt' the file structure by copying/moving files around. I deleted the app and reinstalled again. After that I could import the audio files as you described above.

> Hmmmm I think I get your point, do you mean when transferring inside the Processed folder, for the app to automatically create the items, and well in that case we wouldn't really need a Processed folder, and just use the Documents folder of the app

This is exactly the idea I had. I managed all my audiobooks inside folders. So for me there are no libraries or collections. Basically libraries are folders on the first level and collections are folders inside the top level folders. Therefore an audiobook is just a folder with all the audio files as chapters. 

On my current android device I can manage my audiobooks very easily with this approach. Migrating audiobooks to another android device is just a matter of putting the micro sd card into the new device an configure the 'source' directoy in the audiobook application app.

I would imagine that this would also be possible with iOS. With iPhone 15 I could just connect a USB-C memory stick and copy all files over via the native 'Files' app.

However I understand that my use case is not the common approach. I could live with the current state but wish for the following changes:
- only display playable files (e.g. hide files like .jpg, .json, .txt, etc.)
- display cover.jpg or folder.jpg as album art


Anyway, thanks for the great work on this app. I'll surely send you a tip once the migration to iPhone has been done ;-)

**Comment by raxelbrof** [comment]
Just wondering if any more has come of this, or is likely to come of this. I keep all my audiobooks as folders of mp3 file chapters, and each book folder is in an author folder. It would be great if I could just drag that folder structure into Bookplayer's finder-exposed storage, and for that to just be the library structure that's on the device, no processing or management or anything.

I recreated my on-device library from scratch a few days ago. It took 5 hours. Most of that is just the process of making folders, putting them where you want them to be, importing files, making sure the get moved to the right folders etc.

**Reply by hdo** [reply]
Well, I have given up on this. I use BookPlayer on an occasional basis because it's a hassle with the iOS limitations (file download, share, etc.). After my current mobile plan run out I'll happily switch back to Android :-)

**Reply by GianniCarlo** [reply]
posting here for visibility only, but if you have a Mac, from the Finder, and your phone connected, you can just drop in the folder with the internal structure, and BookPlayer will import it as is (this should work too from iTunes on a Windows). 

https://github.com/user-attachments/assets/678384e9-c64e-41d8-9373-90299d26098c

**Reply by hdo** [reply]
Thanks for the tipp @GianniCarlo! Well, I don't like iTunes that much and don't have it installed either. My workflow is basically this:

- Audiobook Collection is stored on my NAS (Synology)
- Download to Phone via DS File app (works great for me)
- Move Content with iOS Files app from DS File to BookPlayer
- Update BookPlayer Library

It works somehow but due to iOS limitations the experience is kinda lacking.
